presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveWindowQueries.java

```diff
@@ -84,4 +85,101 @@ public void testRowNumber()
+        List<String> overClauses = Arrays.asList("PARTITION BY orderkey ORDER BY commitdate asc nulls first, extendedprice asc, shipinstruct desc",
+                "PARTITION BY commitdate ORDER BY orderkey desc, extendedprice asc, shipinstruct desc nulls first",
+                "PARTITION BY extendedprice ORDER BY commitdate asc nulls first, orderkey desc nulls first, shipinstruct asc nulls first",
+                "PARTITION BY shipinstruct ORDER BY commitdate, extendedprice, orderkey",
+                "ORDER BY commitdate desc, orderkey asc, extendedprice desc nulls first, shipinstruct asc nulls first",
+                "ORDER BY orderkey desc, commitdate asc, extendedprice asc nulls first, shipinstruct desc nulls first",
+                "ORDER BY extendedprice asc nulls first, shipinstruct desc nulls first, commitdate desc, orderkey asc",
+                "ORDER BY shipinstruct desc, orderkey asc, commitdate desc nulls first, extendedprice asc nulls first");
+
+        List<String> frameClauses = Arrays.asList("RANGE UNBOUNDED PRECEDING",
+                "RANGE CURRENT ROW",
+                "RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING",
+                "RANGE BETWEEN CURRENT ROW AND UNBOUNDED FOLLOWING",
+                "ROWS UNBOUNDED PRECEDING",
+                "ROWS CURRENT ROW",
+                "RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING",
+                "ROWS BETWEEN CURRENT ROW AND UNBOUNDED FOLLOWING");
+
+        for (String columnProjection : columnProjections) {
+            for (String overClause : overClauses) {
+                for (String frameClause : frameClauses) {
+                    queries.add(String.format("SELECT %s, %s OVER (%s, %s %s) FROM lineitem", columnProjection, functionCall.first(), overClause, functionCall.second(), frameClause));
+                }
+            }
+        }
+        return queries.build();
+    }
+
+    @Test
+    public void testNthValue()
+    {
+        List<Pair<String, String>> nthValueFunctionPairs = Arrays.asList(Pair.of("nth_value(receiptdate, 5)", "receiptdate"),
+                Pair.of("nth_value(partkey, 5)", "partkey"),
+                Pair.of("nth_value(shipmode, 5)", "shipmode"),
+                Pair.of("nth_value(tax, orderkey + 1)", "tax"),
+                Pair.of("nth_value(linenumber, orderkey + 1)", "linenumber"));
+
+        for (Pair<String, String> nthValueFunctionPair : nthValueFunctionPairs) {
+            List<String> queries = getWindowTestQueries(nthValueFunctionPair);
+            for (String query : queries) {
+                assertQuery(query);
+            }
+        }
+    }
+
+    protected void testAggregateWindowFunction(String aggregateFunction, Boolean numericAggregate)
+    {
+        ImmutableList.Builder<Pair<String, String>> aggregateFunctionPairs = ImmutableList.builder();
+        aggregateFunctionPairs.add(Pair.of(String.format("%s(partkey)", aggregateFunction), "partkey"));
```

Comment:
What is different between partkey and linenumber (in terms of datatype) ? We could reduce to a single test.